### DISCUSSION TITLE
Layout inheritance

### DISCUSSION THREAD

**Discussion by ror3d** [statiqdev/Discussions#112] [Q&A] [answered]
Some static generator templating engines include support for inheriting layouts. Basically, you would have files such as:

`base.cshtml`:

    <html>
    ...
      @RenderBody()
    ...
    </html>

`post.cshtml`:

    ---
    Layout: base.cshtml
    ---
    ... some layout specialization ...
    @RenderBody()

But it doesn't seem to work. Is there some way I haven't found just because of different naming or is it just not possible?

Thanks!

**Comment by daveaglick** [comment, marked as answer]
It's definitely possible - nested layouts are built into the Razor engine and work the same way in Statiq as they do in ASPNET. So then the question becomes why they aren't working in your case.

You can see an example of nested layouts in the Statiq web site here: https://github.com/statiqdev/statiqdev.github.io/tree/develop/input/news

Notice how there's a `_ViewStart.cshtml` file that tells every page under the "news" folder to use the `/news/_layout.cshtml` layout. In turn, the `/news/_layout.cshtml` contains this block at the top:
```
@{
    Layout = @"/_layout.cshtml";
}
```
That instructs the `/news/_layout.cshtml` to nest inside the `/_layout.cshtml` layout.

I wonder if part of the confusion here might be how the `Layout` front matter works inside a layout - the short answer is that it doesn't, which is why the nesting in the example above is done using the `@{ ... }` Razor syntax. The longer explanation is that front matter is only processed for pages being _directly_ rendered by Statiq. Since a layout is something that Statiq doesn't normally process directly, any front matter in a layout file won't have an impact (nor should Statiq deal with layouts - layouts for any given rendering engine are going to be a function of that engine and shouldn't be rendered or treated as renderable content otherwise).

Does that make sense?

**Reply by ror3d** [reply]
Ah indeed! yes, the issue here was the front-matter syntax. In content cshtml files I have the yml style working, but in this case it just got rendered as normal text. When using the `@{ ... }` syntax for it instead it works perfectly. Thank you!
